iidm/iidm-api/src/main/java/com/powsybl/iidm/network/util/DanglingLineBoundaryImpl.java

```diff
@@ -25,27 +25,42 @@ public DanglingLineBoundaryImpl(DanglingLine parent) {
 
     @Override
     public double getV() {
+        if (valid(parent.getP0(), parent.getQ0())) {
+            DanglingLineData danglingLineData = new DanglingLineData(parent, true);
+            return danglingLineData.getBoundaryBusU();
+        }
+
         Terminal t = parent.getTerminal();
         Bus b = t.getBusView().getBus();
         return new SV(t.getP(), t.getQ(), getV(b), getAngle(b), Branch.Side.ONE).otherSideU(parent, true);
     }
 
     @Override
     public double getAngle() {
+        if (valid(parent.getP0(), parent.getQ0())) {
+            DanglingLineData danglingLineData = new DanglingLineData(parent, true);
+            return Math.toDegrees(danglingLineData.getBoundaryBusTheta());
+        }
         Terminal t = parent.getTerminal();
         Bus b = t.getBusView().getBus();
         return new SV(t.getP(), t.getQ(), getV(b), getAngle(b), Branch.Side.ONE).otherSideA(parent, true);
     }
 
     @Override
     public double getP() {
+        if (valid(parent.getP0(), parent.getQ0())) {
```

Comment:
If we have explicit information about boundary injection, we use it. If it is not valid, we rely on flow and voltage at network side to calculate boundary values.